packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -1042,6 +1036,24 @@ const runGenerator = async (
         // Install dependencies for the newly minted project.
         npmInstall(outputDir, {verbose, projectName: context.answers.project.name})
     }
+
+    // Add selected Application Extensions and their default config
+    updatePackageJson(p.resolve(outputDir, 'package.json'), {
+        ...(selectedAppExtensions.length > 0 && {
+            mobify: {
+                app: {
+                    extensions: selectedAppExtensions.map((extension) => {
+                        // Since we've just installed the dependencies, we can read the default config of each extension
+                        // TODO: btw with extracted extensions, why they also appears inside node_modules? So the following line surprisingly works.
+                        const defaultConfig = readJson(
+                            p.join(outputDir, 'node_modules', extension, 'config', 'default.json')
+                        )
```

Comment:
What are we doing in the case where an extension doesn't have a default config? Until we have extension validation we might want to assume that it might be missing and we should put an empty object and warn. (I know we are the only developers of extensions, but that might not be the case in the future)

Reply:
Ok, now if the default config is missing, we'll print out a warning message and also generate a minimal default config `{enabled: true}`.

![iTerm2 2024-12-16 at 12 25 14](https://github.com/user-attachments/assets/4d16d65e-1b9a-4ee7-96dd-594deec52b59)